src/focus.service.ts

```diff
@@ -1,20 +1,17 @@
-import { Location } from '@angular/common';
 import { Injectable } from '@angular/core';
 import { Subscription } from 'rxjs/Subscription';
 
 import 'rxjs/add/operator/filter';
 
 import { ArcEvent } from './event';
+import { findElByDistance } from './focus-strategies/focus-by-distance';
 import { FocusByRegistry } from './focus-strategies/focus-by-registry';
 import { Direction, isHorizontal } from './model';
 import { RegistryService } from './registry.service';
 
 const defaultFocusRoot = document.body;
 // These factors can be tweaked to adjust which elements are favored by the focus algorithm
 const scoringConstants = Object.freeze({
-  primaryAxisDistanceWeight: 30,
```

Comment:
Thinking about it more, I am worried about getting rid of the history & shadow stuff... specifically without anything to replace it. I can explain more when you're back Saadat.